simple-facebook-instant-articles.php

```diff
@@ -403,7 +406,41 @@ public function api_galleries_shortcode( $atts ) {
 	 */
 	public function reformat_social_embed( $html, $url, $attr, $post_ID = null ) {
 
-		return '<figure class="op-social"><iframe>' . $html . '</iframe></figure>';
+		return sprintf( '<figure class="op-social"><iframe>%s</iframe></figure>', $html );
+	}
+
+	/**
+	 * Ensure brightcove scripts are loaded.
+	 *
+	 * @param  string $html Embed markup.
+	 * @param  string $url  Embed url.
+	 *
+	 * @return string Embed markup.
+	 */
+	function load_brightcove_scripts( $html, $url ) {
+		global $wp_embed;
+
+		$brightcove_handler = null;
+
+		// Find brightcove embed handler.
+		foreach ( $wp_embed->handlers as $priority => $handlers ) {
+			foreach ( $handlers as $name => $args ) {
+				if ( 'brightcove' === $name ) {
+					$brightcove_handler = $args;
+					break;
+				}
+			}
+		}
+
+		// If we've found the brightcove handler, and the regex matches, output the brightcove script.
+		if ( $brightcove_handler && preg_match( $brightcove_handler['regex'], $url ) ) {
+			ob_start();
+			do_action( 'wp_enqueue_scripts' );
```

Comment:
@mattheu can you explain why do we need `do_action( 'wp_enqueue_scripts' );` here?